### PR TITLE
Disallow multiple targets with the same name

### PR DESCRIPTION
**Pull request by dchw**
This is more or less a small validation framework we can grow. Example output:

image

### CODE REVIEW

**Comment by alexcb** [comment]
This seems like a nice to have feature; is it possible to add a test for this?

File: earthfile2llb/validator.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package earthfile2llb
```
Comment (vladaionescu) [inline earthfile2llb/validator.go:1]: Could this file go under the `ast` package and have public functions?

File: earthfile2llb/earthfile2llb.go [hunk 2]
```diff
@@ -115,6 +115,9 @@ func Earthfile2LLB(ctx context.Context, target domain.Target, opt ConvertOpt) (m
 	if err != nil {
 		return nil, err
 	}
+	if err := validateAst(bc.Earthfile); err != nil {
+		return nil, err
+	}
```
Comment (vladaionescu) [inline earthfile2llb/earthfile2llb.go:120]: I would put this check in the build context resolver, right after the parse call. That way  `bc` always has a validated `bc.Earthfile`.

File: earthfile2llb/validator.go (hunk 1 above)
Comment (dchw) [reply earthfile2llb/validator.go:1]: Yeah; I had started here hoping to do a little more but decided it wasnt worth it. Ill move it to finish formalizing that

File: earthfile2llb/earthfile2llb.go (hunk 2 above)
Comment (dchw) [reply earthfile2llb/earthfile2llb.go:120]: How about putting it in the `Parse` method itself? Then its always valid after parsing, and we don't have to make validation public?

**Comment by dchw** [comment]
> This seems like a nice to have feature; is it possible to add a test for this?

Not just nice to have; we are allowing duplicate target names within a file right now :(

File: earthfile2llb/earthfile2llb.go (hunk 2 above)
Comment (vladaionescu) [reply earthfile2llb/earthfile2llb.go:120]: That might be better

**Comment by dchw** [comment]
@alexcb Test added.

**Review by alexcb — APPROVED** [review]

**Pull request merged**